test_data/std/permutation_via_challenges.asm

```diff
@@ -0,0 +1,21 @@
+use std::prover::Query;
+use std::convert::fe;
+use std::permutation::permutation;
+
+machine Main {
+    degree 8;
+
+    col fixed first_four = [1, 1, 1, 1, 0, 0, 0, 0];
+
+    // Two pairs of witness columns, claimed to be permutations of one another
+    // (when selected by first_four and (1 - first_four), respectively)
+    col witness a1(i) query Query::Hint(fe(i));
+    col witness a2(i) query Query::Hint(fe(i + 42));
+    col witness b1(i) query Query::Hint(fe(7 - i));
+    col witness b2(i) query Query::Hint(fe(7 - i + 42));
+
+    // TODO: Functions currently cannot add witness columns, so we have to manually
```

Comment:
I think they can, @chriseth didn't you add this recently?

Reply:
They can but only at the current stage.

Reply:
Although the other PR also has `new_witness_in_stage`: https://github.com/powdr-labs/powdr/pull/1195/files#diff-ca3855de0b1297c7047e224cca08ed50dd7e5cdc5c9d3938c402c35f03d1f45fR23